src/mutations/appData.ts

```diff
@@ -75,10 +87,46 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
       }
     },
     onSettled: () => {
-      const data = getData(queryClient);
-      queryClient.invalidateQueries(buildAppDataKey(data?.itemId));
+      if (!enableWebsocket) {
+        const data = getData(queryClient);
+        queryClient.invalidateQueries(buildAppDataKey(data?.itemId));
+      }
     },
   });
+
+  /**
+   * @description
+   * By using the `merge` method in the `onMutate` callback, the payload should be a clean object, free of attributes from the `Record` type
+   * from Immutable.js. Therefore, when using this mutation, one must be careful not to use object spreading
+   * of the `data` attribute found in the `AppDataRecord`. I believe this may break some apps.
+   *
+   * @example Working example
+   * ```
+   * const { id, data } = appData; // type: AppDataRecord
+   * // data: { text: string, count: number }
+   * patchAppData({
+   *  id,
+   *  data: {
+   *    count: data.count,
+   *    text: newContent,
+   *  },
+   * });
+   * ```
+   *
+   * @example Failing example
+   * ```
+   * const { id, data } = appData; // type: AppDataRecord
+   * // data: { text: string, count: number }
+   * patchAppData({
+   *  id,
+   *  data: {
+   *    ...data,
+   *    text: newContent,
+   *  },
+   * });
+   * ```
+   *
+   */
```

Comment:
Nice doc! :book:

Reply:
I was stuck with the failing example for a while... 😬